src/role-access/containers/exclusion/exclusion-home/exclusion-home.component.ts

```diff
@@ -0,0 +1,145 @@
+    );
+  }
+
+  public onNavEvent(event: ExclusionNavigationEvent) {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
+    };
+    this.navigationHandler(event);
+  }
+
+  public isComponentVisible(currentNavigationState: ExclusionState, requiredNavigationState: ExclusionState[]): boolean {
+    return requiredNavigationState.includes(currentNavigationState);
+  }
+
+  public navigationHandler(navEvent: ExclusionNavigationEvent) {
+    switch (navEvent) {
+      case ExclusionNavigationEvent.BACK: {
+        switch (this.navigationCurrentState) {
+          case ExclusionState.CHOOSE_EXCLUSION:
+            // TODO back to case detail tabs
+            // this.router.navigateByUrl(`cases/case-details/${caseId}`).then(r => {
+            //   return;
+            // });
+            break;
+          case ExclusionState.CHOOSE_PERSON_ROLE:
+            this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.CHOOSE_EXCLUSION));
+            break;
+          case ExclusionState.FIND_PERSON:
+            this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.CHOOSE_PERSON_ROLE));
+            break;
+          case ExclusionState.DESCRIBE_EXCLUSION:
+            // TODO if from exclude me
+            // this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.CHOOSE_EXCLUSION));
+            // TODO if from exclude another person
+            // this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.FIND_PERSON));
+            break;
+          case ExclusionState.CHECK_ANSWERS:
+            this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.DESCRIBE_EXCLUSION));
+            break;
+          default:
+            throw new Error('Invalid exclusion state');
+        }
+        break;
+      }
+      case ExclusionNavigationEvent.CONTINUE: {
+        switch (this.navigationCurrentState) {
+          case ExclusionState.CHOOSE_EXCLUSION:
+            this.chooseExclusionComponent.navigationHandler(navEvent);
+            break;
+          case ExclusionState.CHOOSE_PERSON_ROLE:
+            this.choosePersonRoleComponent.navigationHandler(navEvent);
+            break;
+          case ExclusionState.FIND_PERSON:
+            this.findPersonComponent.navigationHandler(navEvent);
+            break;
+          case ExclusionState.DESCRIBE_EXCLUSION:
+            this.describeExclusionComponent.navigationHandler(navEvent);
+            break;
+          default:
+            throw new Error('Invalid exclusion state');
+        }
+        break;
+      }
+      case ExclusionNavigationEvent.CONFIRM_EXCLUSION: {
+        switch (this.navigationCurrentState) {
+          case ExclusionState.CHECK_ANSWERS:
+            // TODO
+            break;
+          default:
+            throw new Error('Invalid exclusion state');
+        }
+        break;
+      }
+      default:
+        throw new Error('Invalid exclusion navigation event');
```

Comment:
Do we need to throw any error?

Reply:
yes we got much benefit from it when development if not define the correct navigation, but it's okay to remove it when go live